src/engine/uct/node/mod.rs

```diff
@@ -146,21 +146,38 @@ impl Node {
                 .iter()
                 .map(|&m| Node::new(m, self.config))
                 .collect();
+            if self.children.len() <= (game.size() * game.size() / 10) as usize {
+                let player = game.last_move().color().opposite();
+                if game.winner() == player || !self.config.uct.playout_aftermath { //don't pass if we're losing on the board
+                    self.children.push(Node::new(Pass(player), self.config));
+                }
+            }
+            
             self.descendants = self.children.len();
         }
  }
 
     pub fn expand(&mut self, board: &Board) -> bool {
         let not_terminal = !board.is_game_over();
-        let mut children = vec![];
         if not_terminal && self.plays >= self.config.uct.expand_after {
-            children = board.legal_moves_without_eyes()
+            let mut children = board.legal_moves_without_eyes()
                 .iter()
                 .map(|m| self.new_leaf(board, m))
                 .collect();
+                
+            self.priors(&mut children, board);
+            self.children = children;
+            
+            if self.children.len() <= (board.size() * board.size() / 10) as usize {
+                let player = board.next_player();
+                if board.winner() == player { //don't pass if we're losing on the board
```

Comment:
Well we need to be able to see our opponent passing in the game tree. For example, let's say it's a ten thousand year ko. If we make our opponent play self-atari with 4 stones (usually allowed, but bad in this case because it makes our stones live) then we just assume we win the game. But in fact if our opponent passes, we can't win the ko if we don't have ko threats.